### PR TITLE
finish task5

### PR DESCRIPTION
**Pull request by li-wang-uk**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -85,4 +86,28 @@ describe( "/api/articles/:article_id", ()=> {
...
+            body.articles.forEach((article)=> {
+            expect(typeof article.author).toBe("string")
+            expect(typeof article.title).toBe("string")
+            expect(typeof article.article_id).toBe("number")
+            expect(typeof article.topic).toBe("string")
+            expect(typeof article.created_at).toBe("string")
+            expect(typeof article.votes).toBe("number")
+            expect(typeof article.article_img_url).toBe("string")
+            expect(typeof Number(article.comment_count)).toBe("number")
+            expect(Object.keys(article).includes('body')).toBe(false)
```
Comment (popatre) [inline __tests__/app.test.js:108]: i wouldnt change the data type of your response then check it - if you want it to be a number, you need to make sure you've sent it back as a number

File: api/models/articles-models.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            return Promise.reject({ status: 404, msg: "Article Not Found" });
+        }
+        return result.rows
+    })
+  }
+
+  exports.selectAllArticles = () => {
+    const queryString = `SELECT COUNT(c.comment_id) AS comment_count, a.author, a.title, a.article_id, topic, a.created_at, a.votes, a.article_img_url FROM comments c
+    FULL OUTER JOIN articles a 
+    ON a.article_id = c.article_id
```
Comment (popatre) [inline api/models/articles-models.js:21]: nice psql queries here

File: api/models/articles-models.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    .then((result) => {
+        if (result.rows.length === 0){
+            return Promise.reject({ status: 404, msg: "Article Not Found" });
+        }
+        return result.rows
+    })
+  }
+
+  exports.selectAllArticles = () => {
+    const queryString = `SELECT COUNT(c.comment_id) AS comment_count, a.author, a.title, a.article_id, topic, a.created_at, a.votes, a.article_img_url FROM comments c
```
Comment (popatre) [inline api/models/articles-models.js]: you'd make sure comment count is a number here -instead of changing the tests

**Review by popatre — CHANGES_REQUESTED** [review]
Please just change that one test - dont change the data types in the test 

 if you want it to be a number, you need to make sure you've sent it back as a number

File: __tests__/app.test.js [hunk 4]
```diff
@@ -104,10 +104,11 @@ describe( "/api/articles", ()=> {
             expect(typeof article.created_at).toBe("string")
             expect(typeof article.votes).toBe("number")
             expect(typeof article.article_img_url).toBe("string")
-            expect(typeof Number(article.comment_count)).toBe("number")
+            expect(typeof article.comment_count).toBe("number")
```
Comment (ghost) [inline __tests__/app.test.js:107]: For testing if the body is not there, a quicker way would be to check if the ```article.hasOwnProperty('body')``` is false. Object.keys spends time iterating through the object to make an array of key elements, then to use includes does another iteration to inspect that array for the item you are looking for. A slightly different method will reduce two iterations.

**Review by ghost — APPROVED** [review]
Better! Never coerce data types in your test if that is what you are looking for.

**Pull request merged**